app/src/main/java/net/gsantner/markor/model/Document.java

```diff
@@ -99,6 +105,16 @@ public File getFile() {
         return _file;
     }
 
+    public long lastModified() {
+        try {
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+                return Files.readAttributes(_file.toPath(), BasicFileAttributes.class).lastModifiedTime().toMillis();
```

Comment:
This method seems much more consistent than `lastModified`

Reply:
is there also such a method in SAF available?

Reply:
I don't see it. But `DocumentFile.lastModified()` may be more reliable?

Reply:
I'd not further delay this PR. But for future I'd like try it and use it if it works 😄

Reply:
I'd not further delay this PR. But for future I'd like try it and use it if it works smile